sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodegenExpressionCachingSuite.scala

```diff
@@ -85,6 +85,36 @@ class CodegenExpressionCachingSuite extends SparkFunSuite {
     assert(instance2.eval(null))
   }
 
```

Comment:
Just a suggestion; Is this file a right place to add this kind of tests? Looked around the related suite files in `expressions/codegen/`, I couldn't find existing tests for common subexpr elimination in `GenerateMutableProjection` and `GenerateUnsafeProjection`. So, why don't you create a new test suite (`CodegenSubexpressionEliminationSuite` or something) and then add this test there? To improve test coverage, it might be better to add tests having the same granularity for the two projections in a separate PR.

Reply:
Ok, sounds good.

Reply:
Thanks!